### PR TITLE
AAD authentication for EventHubs implemented for MWS

### LINKED ISSUE #845: Support SP AAD tokens to authenticate against EventHubs for MWS
Now MWS deployment don't support AAD tokens to authenticate against EH. 

Config.csv/delta config should have a column which will be used to support AAD tokens to authenticate agains EH.
Below code is an example of connecting through AAD tokens.
      val basePath = s"$tempPath/$etlDBName"
      val azureLogConfig = AzureAuditLogEventhubConfig(connectionString = evhubConnString, eventHubName = 
     "evhName", auditRawEventsPrefix = basePath,
        azureClientId = Some("<service-principal-client-id>"), // this could be also specified as {{secrets/<scope>/<key>}}
        azureClientSecret = Some("{{secrets/<scope>/<secret>}}"),
        azureTenantId = Some("<azure tenant ID>")
      )

### PR DESCRIPTION
**Pull request by sriram251-code**
closes #845 
Test notebook "PR_872_AAD_MWS_TEST_720" added in PR test folder.

### CODE REVIEW

**Review by alexott — CHANGES_REQUESTED** [review]
as was discussed in another PR, we need to add support for different Azure clouds (make it optional at least).

**Review by alexott — APPROVED** [review]
lgtm

File: src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala [hunk 1]
```diff
@@ -490,51 +530,22 @@ object DeploymentValidation extends SparkSessionWrapper {
...
-                                scope: String,
-                                optKey: Option[String],
-                                optEHName: Option[String],
-                                outputPath: String
+                               config: MultiWorkspaceConfig
                               ): DeploymentValidationReport = {
-    if (optKey.isEmpty || optEHName.isEmpty) throw new BadConfigException("When cloud is Azure, the eh_name and " +
-      "eh_scope_key are required fields but they were empty in the config")
+
+    val isAAD = config.aad_client_id.nonEmpty && config.aad_tenant_id.nonEmpty && config.aad_client_secret_key.nonEmpty
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala:536]: I think you're missing validation of the composite sets.

You need to validate that the config provides one of the two following states and derive isAAD from there -- you should create a stand-alone function for this and provide sufficient unit tests for positive / negative outcomes. If user violates these required conditions -- throw a user friendly error.

isEHAAD:
--- nonEmpty ---
aad_client_id
aad_tenant_id
aad_client_secret_key
eh_name
--- empty ---
eh_scope_key

!isEHAAD
--- nonEmpty ---
eh_scope_key
eh_name
--- empty --- 
aad_client_id
aad_tenant_id
aad_client_secret_key

Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala]: done

**Review by GeekSheikh — APPROVED** [review]

**Pull request merged**